src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
+        _dirtyMsaaSamplingPattern = true;
 
-    OsEng.renderEngine().raycasterManager().removeListener(*this);
-    OsEng.renderEngine().deferredcasterManager().removeListener(*this);
-}
+        glBindFramebuffer(GL_FRAMEBUFFER, defaultFbo);
 
-void FramebufferRenderer::raycastersChanged(VolumeRaycaster&, bool) {
-    _dirtyRaycastData = true;
-}
+        _resolveProgram = ghoul::opengl::ProgramObject::Build(
+            "Framebuffer Resolve",
+            absPath("${SHADERS}/framebuffer/resolveframebuffer.vert"),
+            absPath("${SHADERS}/framebuffer/resolveframebuffer.frag")
+        );
 
-void FramebufferRenderer::deferredcastersChanged(Deferredcaster& deferredcaster,
-                                                 isAttached isAttached)
-{
-    (void) deferredcaster;
-    (void) isAttached;
-    _dirtyDeferredcastData = true;
-}
+        _uniformCache.mainColorTexture = _resolveProgram->uniformLocation("mainColorTexture");
+        _uniformCache.blackoutFactor = _resolveProgram->uniformLocation("blackoutFactor");
+        _uniformCache.nAaSamples = _resolveProgram->uniformLocation("nAaSamples");
 
-void FramebufferRenderer::update() {
-    if (_dirtyResolution) {
-        updateResolution();
-        updateMSAASamplingPattern();
+        OsEng.renderEngine().raycasterManager().addListener(*this);
+        OsEng.renderEngine().deferredcasterManager().addListener(*this);
     }
 
-    if (_dirtyRaycastData) {
-        updateRaycastData();
+    void FramebufferRenderer::deinitialize() {
+        LINFO("Deinitializing FramebufferRenderer");
+
+        glDeleteFramebuffers(1, &_mainFramebuffer);
+        glDeleteFramebuffers(1, &_exitFramebuffer);
+        glDeleteFramebuffers(1, &_deferredFramebuffer);
+
+        glDeleteTextures(1, &_mainColorTexture);
+        glDeleteTextures(1, &_mainDepthTexture);
+
+        // DEBUG: deferred g-buffer
+        glDeleteTextures(1, &_deferredColorTexture);
+        glDeleteTextures(1, &_mainPositionTexture);
+        glDeleteTextures(1, &_mainNormalTexture);
+
+        glDeleteTextures(1, &_exitColorTexture);
+        glDeleteTextures(1, &_exitDepthTexture);
+
+        glDeleteBuffers(1, &_vertexPositionBuffer);
+        glDeleteVertexArrays(1, &_screenQuad);
+
+        OsEng.renderEngine().raycasterManager().removeListener(*this);
+        OsEng.renderEngine().deferredcasterManager().removeListener(*this);
     }
 
-    if (_dirtyDeferredcastData) {
-        updateDeferredcastData();
+    void FramebufferRenderer::raycastersChanged(VolumeRaycaster&, bool) {
+        _dirtyRaycastData = true;
     }
 
-    // If the resolve dictionary changed (or a file changed on disk)
-    // then rebuild the resolve program.
-    if (_hdrBackGroundProgram && _hdrBackGroundProgram->isDirty()) {
-        _hdrBackGroundProgram->rebuildFromFile();
+    void FramebufferRenderer::deferredcastersChanged(Deferredcaster& deferredcaster,
+        isAttached isAttached)
+    {
+        (void)deferredcaster;
```

Comment:
If the variables are not used, better to delete their names from the function definition